app/models/thredded/messageboard.rb

```diff
@@ -3,11 +3,13 @@ class Messageboard < ActiveRecord::Base
     FILTERS = %w(markdown bbcode)
 
     extend FriendlyId
-    friendly_id :name, use: :slugged
+    friendly_id :slug_candidates,
+                use:            [:slugged, :reserved],
+                # Avoid route conflicts
+                reserved_words: %w(messageboards private-topics autocomplete-users theme-preview)
```

Comment:
I like how you deferred `slug_candidates` to the private method below, returning that array. What about doing the same with this array of possible route conflicts?

```ruby
# ...
reserved_words: possible_route_conflicts

private

def possible_route_conflicts
  %w(messageboards private-topics autocomplete-users theme-preview)
end
```